bears/general/InvalidLinkBear.py

```diff
@@ -49,8 +49,33 @@ def parse_pip_vcs_url(link):
     def find_links_in_file(file, timeout, link_ignore_regex):
         link_ignore_regex = re.compile(link_ignore_regex)
         regex = re.compile(
-            r'((git\+|bzr\+|svn\+|hg\+|)https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s('
-            r')%\'"`<>|\\]+|\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
+            r"""
+            ((git\+|bzr\+|svn\+|hg\+|)  # For VCS URLs
+            https?://                   # http:// or https:// as only these
+                                        # are supported by the ``requests``
+                                        # library
+            [^.:%\s_/?#[\]@\\]+         # Initial part of domain
+            \.                          # A required dot `.`
+            (
+                (?:[^\s()%\'"`<>|\\]+)  # Path name
+                                        # This part does not allow
+                                        # any parenthesis: balanced or
+                                        # unbalanced.
+            |                           # OR
+                \([^\s()%\'"`<>|\\]*\)  # Path name contained within ()
```

Comment:
Question here: Why do we assume there can't be unbalanced paranthesis in a URL ?